### ISSUE TITLE
eval script to offset a tensor to the next or previous positions inside a 1vec?

### ISSUE THREAD

**Issue by dza6549** [tkalayci71/embedding-inspector#18]
Hi

Would it be possible to offset the tensor N number of places forward or backward? So rather than 000100 we could get (next + 1) 000010 or (next - 1) 001000 and move the position of the tensor to the right or to the left N number of places.

In a psuedocode array I'd do something like 

For int i to 768
Val(i) = Val(i+1) or Val(i) = Val(i-1)

Maybe we'd have to deal with tensor 0 and tensor 768 - but my experiments with your extension seem to show that the embedding is pretty forgiving if you put 100 tensors in the 700-768 position and doesn't seem to complain about the left over tensors, 100-68 = 32 tensors being out of bounds.

Hope this makes sense. Please let me know if I can provide any further info.

Cheers and thanks

**Comment by tkalayci71** [comment]
Hi, yes it is possible, to explain in steps:

```
this eval will change nothing:
= tot_vec[n,i]

this eval will shift values, but will give error
= tot_vec[n,i+100]

so take mod of the index to avoid error, but now values will be rotated
= tot_vec[n,(i+100)%maxi]

so multiplying with range check will do what you ask:
= tot_vec[n,(i+100)%maxi] * ( (i+100)>=0) * ( (i+100)<maxi )

to change the shift amount, replace all 3 copies of "i+100" to like "i+200" or "i-300"

```
Cheers!

**Comment by dza6549** [comment]
Thank you @tkalayci71 !

I had got to = tot_vec[n,i-1] but your script is much more robust (of course).

Cheers and thanks!

PS I used your bit script to make slices of a character embedding vector with only 50 tensors 

`slc50-100','=v*(i>=50)*(i<100)`

And the image still produces the character which is remarkable. I hypothesise that the 1vec is symbolically 'cat in house, cat running a race, cat eating food, cat jumping, cat sleeping, etc' But of course my human brain is looking for an answer in human terms and I am not sure the model sees the world the same way.
